MCSync/frmMain.cs

```diff
@@ -14,95 +23,77 @@ public frmMain()
-        }
+            try
+            {
+                int rand = new Random().Next(1, 105);
 
-        private void newBtnSync_MouseEnter(object sender, EventArgs e)
-        {
-            btnSync.BackgroundImage = Properties.Resources.btnSyncOver;
-        }
+                var request = WebRequest.Create("https://mesabrook.com/backgrounds/background" + rand + ".png");
 
-        private void newBtnSync_MouseLeave(object sender, EventArgs e)
-        {
-            btnSync.BackgroundImage = Properties.Resources.btnSyncBase;
+                using (var response = request.GetResponse())
+                using (var stream = response.GetResponseStream())
+                {
+                    BackgroundImage = Bitmap.FromStream(stream);
+                    BackgroundImageLayout = ImageLayout.Stretch;
+                }
+            }
+            catch (Exception ex)
+            {
+                BackgroundImage = Properties.Resources.b1;
+            }
         }
 
-        private void newBtnConf_MouseClick(object sender, MouseEventArgs e)
+        private void fancyButton1_Click(object sender, EventArgs e)
         {
+            pnlMain.Hide();
             frmConfig config = new frmConfig();
-            config.ShowDialog();
-        }
 
-        private void newBtnSync_MouseClick(object sender, MouseEventArgs e)
-        {
-            frmSync sync = new frmSync();
-            sync.Show();
-            Hide();
+            config.TopLevel = false;
+            config.ControlBox = false;
+            config.Dock = DockStyle.Fill;
+            config.FormBorderStyle = FormBorderStyle.None;
+
+            pnlForm.Controls.Add(config);
+            config.Show();
         }
 
-        private void btnHelp_MouseClick(object sender, MouseEventArgs e)
+        private void frmMain_HelpButtonClicked(object sender, System.ComponentModel.CancelEventArgs e)
         {
             Process.Start("https://mesabrook.com/mcsync/index.html");
         }
 
-        private void btnHelp_MouseEnter(object sender, EventArgs e)
+        private void fButtonSync_Click(object sender, EventArgs e)
         {
-            lblDocumentation.Visible = true;
-        }
+            frmSync sync = new frmSync();
+            sync.TopLevel = false;
+            sync.ControlBox = false;
+            sync.Dock = DockStyle.Fill;
+            sync.FormBorderStyle = FormBorderStyle.None;
 
-        private void btnHelp_MouseLeave(object sender, EventArgs e)
-        {
-            lblDocumentation.Visible = false;
+            pnlForm.Controls.Add(sync);
+            sync.Show();
         }
 
-        public void backgroundChooser()
+        private void pnlMainTimer_Tick(object sender, EventArgs e)
         {
-            int bg = new Random().Next(0, 6);
-            switch(bg)
+            var frmConfig = Application.OpenForms["frmConfig"];
+            var frmSync = Application.OpenForms["frmSync"];
```

Comment:
We should use concrete types when we know them (in this case, `Form`).

Also, wouldn't these potentially cause an error if we never opened either of these forms?

Reply:
This code is crappy, but it's vital 🤣🤣🤣🤣

Reply:
Fixed